src/mots/directory.py

```diff
@@ -0,0 +1,216 @@
+            logger.debug("People directory modified, updating configuration...")
+            self.config_handle.config["people"] = self.people.serialized
+
+    def query(self, *paths: str) -> QueryResult:
+        """Query given paths and return a list of corresponding modules."""
+        result = {}
+        rejected = []
+        for path in paths:
+            if not (self.repo_path / path).exists():
+                logger.warning(f"Path {path} does not exist, skipping.")
+                rejected.append(path)
+                continue
+            result[path] = self.index.get(self.repo_path / path, list())
+        logger.debug(f"Query {paths} resolved to {result}.")
+
+        return QueryResult(result, rejected)
+
+
+class QueryResult:
+    """Helper class to simplify query result interpretation."""
+
+    paths: list = None
+    path_map: dict = None
+    rejected_paths: list = None
+    modules: list = None
+    owners: list = None
+    peers: list = None
+
+    data_keys = {
+        "paths",
+        "rejected_paths",
+        "modules",
+        "owners",
+        "peers",
+    }
+
+    def __init__(self, result, rejected):
+        data = {k: set() for k in self.data_keys}
+        self.path_map = result
+
+        for path in self.path_map:
+            data["paths"].add(path)
+            data["modules"].update(self.path_map[path])
+
+        for module in data["modules"]:
+            data["owners"].update([Person(**o) for o in module.owners])
+            data["peers"].update([Person(**p) for p in module.peers])
+
+        data["rejected_paths"].update(rejected)
+
+        for key in data:
+            setattr(self, key, list(data[key]))
+
+    def __add__(self, query_result):
+        """Merge the data from both QueryResult objects."""
+        path_map = self.path_map.copy()
+        path_map.update(query_result.path_map)
+        rejected_paths = set.intersection(
+            set(self.rejected_paths),
+            set(query_result.rejected_paths),
+        )
+        return QueryResult(path_map, rejected_paths)
+
+    def __radd__(self, query_result):
+        """Call self.__add__ since the order of addition does not matter."""
+        return self.__add__(query_result)
+
+
+@dataclass
+class Person:
+    """A class representing a person."""
+
+    bmo_id: int
+    name: str = ""
+    info: str = ""
+    nick: str = ""
+    bmo_data: InitVar[dict] = None
+
+    def __post_init__(self, bmo_data):
+        """Refresh BMO data from BMO API."""
+        if bmo_data:
+            self.nick = bmo_data.get("nick", "")
+            self.bmo_id = bmo_data.get("id", self.bmo_id)
+            real_name = bmo_data.get("real_name", "")
+
+            parsed_real_name = parse_real_name(real_name)
+            self.name = parsed_real_name["name"]
+            self.info = parsed_real_name["info"]
+
+    def __hash__(self):
+        """Return a unique identifier for this person."""
+        return int(self.bmo_id)
```

Comment:
Smart :)
FWIW, `self.bmo_id` should already be an int, right? If `bmo_data["id"]` is a string, we should parse it accordingly so that the data is clean by the time it's represented by this data object.